cmd/skopeo/list_tags_test.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+// Tests the kinds of inputs allowed and expected to the command
+func TestDockerRepositoryReferenceParser(t *testing.T) {
+	for _, test := range [][]string{
+		{"docker://myhost.com:1000/nginx", "myhost.com:1000/nginx"},    //no tag
+		{"docker://myhost.com/nginx", "myhost.com/nginx"},              //no port or tag
+		{"docker://somehost.com", "docker.io/library/somehost.com"},    // Valid default expansion
+		{"docker://nginx", "docker.io/library/nginx"},                  // Valid default expansion
+		{"docker://myhost.com:1000/nginx:foobar:foobar", ""},           // Invalid repository ref
+		{"docker://somehost.com:5000/", ""},                            // no repo
+		{"docker://myhost.com:1000/nginx:latest", ""},                  //tag not allowed
+		{"docker://myhost.com:1000/nginx@sha256:abcdef1234567890", ""}, //digest not allowed
+	} {
+
+		ref, err := parseDockerRepositoryReference(test[0])
+
+		if test[1] == "" {
+			assert.Error(t, err, "Expected error in parsing but no error raised for %v", test[0])
+		} else {
+			if assert.NoError(t, err, "Could not parse, got error on %v", test[0]) {
+				assert.Equal(t, test[1], ref.DockerReference().Name(), "Mismatched parse result for input %v", test[0])
+			}
+		}
+	}
+}
+
+func TestDockerRepositoryReferenceParserDrift(t *testing.T) {
```

Comment:
Non-blocking: An alternative to this approach would be to actually use `alltransports.ParseImageName().DockerReference()` as the primary data in `parseDockerRepositoryReference`, and do an extra `ParseNormalizedNamed` check at run-time only to check there isn’t a tag/digest.

That check can also add an “if the two don’t match, fail with an internal error” condition — that would ensure that they _always_ match even in cases that this forgets to test, and also all tests of `parseDockerRepositoryReference` would implicitly verify that the two results match in all tested cases.